packages/plugin-http-client/src/HttpClient.ts

```diff
@@ -0,0 +1,160 @@
+import type {
+  Dependencies,
+  HttpAgent,
+  HttpAgentRequestOptions,
+  UnknownParameters,
+} from '@ima/core';
+
+declare module '@ima/core' {
+  interface OCAliasMap {
+    HttpClientDefaultProcessors: Processor[];
+  }
+}
+
+import {
+  AbstractProcessor,
+  Processor,
+  Operation,
+  ProcessorParams,
+} from './AbstractProcessor';
+
+export const OPTION_TRANSFORM_PROCESSORS = 'transformProcessors';
+
+export type HttpClientRequestOptions = {
+  [OPTION_TRANSFORM_PROCESSORS]?: (processor: Processor[]) => Processor[];
+} & HttpAgentRequestOptions;
+
+export type HttpClientRequestMethod =
+  | 'get'
+  | 'put'
+  | 'post'
+  | 'patch'
+  | 'delete';
+
+export type HttpClientRequest = {
+  method: HttpClientRequestMethod;
+  url: string;
+  data?: Record<string, any>;
+  options?: HttpClientRequestOptions;
+};
+
+/**
+ * HttpClient use ima HttpAgent and adds support for processors.
+ */
+export class HttpClient {
+  #http: HttpAgent;
+  #defaultProcessors: Processor[];
+
+  static get $dependencies(): Dependencies {
+    return ['$Http', '...?HttpClientDefaultProcessors'];
+  }
+
+  constructor(http: HttpAgent, ...defaultProcessors: Processor[]) {
+    this.#http = http;
+
+    this.#defaultProcessors = defaultProcessors.filter(
+      item => item instanceof AbstractProcessor
+    );
+  }
+
+  /**
+   * Method registerProcessor added to defaultProcessors new processor
+   * @param processor
+   */
+  registerProcessor(processor: Processor) {
+    this.#defaultProcessors.push(processor);
```

Comment:
In constructor (on line 56) you filter out processors which are not instanceOf AbstractProcessor, but here you push anything. Shouldn't you make the same check? You rely on processors having preRequest and postRequest methods so it make sense to make this check I think.

Reply:
I added check for every place where is processor added.